code/backend/batch/utilities/helpers/AzureSearchIVDatasourceHelper.py

```diff
@@ -0,0 +1,47 @@
+from azure.search.documents.indexes.models import (
+    SearchIndexerDataContainer,
+    SearchIndexerDataSourceConnection,
+)
+from azure.search.documents.indexes._generated.models import (
+    NativeBlobSoftDeleteDeletionDetectionPolicy,
+)
+from azure.search.documents.indexes import SearchIndexerClient
+from .EnvHelper import EnvHelper
+from azure.identity import DefaultAzureCredential
+from azure.core.credentials import AzureKeyCredential
+
+
+class AzureSearchIVDatasourceHelper:
+    def __init__(self, env_helper: EnvHelper):
+        self.env_helper = env_helper
+        self.indexer_client = SearchIndexerClient(
+            self.env_helper.AZURE_SEARCH_SERVICE,
+            (
+                AzureKeyCredential(self.env_helper.AZURE_SEARCH_KEY)
+                if self.env_helper.AZURE_AUTH_TYPE == "keys"
```

Comment:
Can we extract `if self.env_helper.AZURE_AUTH_TYPE == "keys"` into a method, possibly in EnvHelper that is called something like, isAuthTypeKeys(), so that we don't leak the exact format of the "keys" string everywhere? Thanks!

Reply:
In fact, it looks like we could also extract this logic of creating the SearchIndexerClient somewhere common for all the places using it.

Reply:
Updated it to call a function to get the Auth_type. SearchIndexerClient is being called only for IV, so I dont think it should be in EnvHelper class. Let me know your thoughts